core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -237,10 +244,19 @@ public Snapshot apply() {
 
     OutputFile manifestList = manifestListPath();
 
+    EncryptionManager encryptionManager = ops.encryption();
+    EncryptedOutputFile encryptedManifestList = encryptionManager.encrypt(manifestList);
+
+    long manifestListSize = 0L;
+    ByteBuffer manifestListKeyMetadata =
+        (encryptedManifestList.keyMetadata() == null)
+            ? null
+            : encryptedManifestList.keyMetadata().buffer();
+
     try (ManifestListWriter writer =
         ManifestLists.write(
             ops.current().formatVersion(),
-            manifestList,
+            encryptedManifestList.encryptingOutputFile(),
```

Comment:
For manifests, we added methods that accepted `EncryptedOutputFile`. I think it makes sense to do the same thing here so that we can delegate more.

Reply:
Per your suggestion below,

>  I don't think I have a strong preference whether this is delegated to the writer (toManifestListFile()) or just done in a method like ManifestListFile.create(EncryptionManager em, String wrappedKey, ByteBuffer keyMetadataBuffer, String location)

Using delegation to ManifestListWriter (`toManifestListFile()` would require passing the encryption manager. I'll implement a static `ManifestListFile.create` method.